service/history/api/consistency_checker.go

```diff
@@ -178,9 +179,91 @@ func (c *WorkflowConsistencyCheckerImpl) getWorkflowLeaseValidatedByClock(
 		release(err)
 		return nil, err
 	}
+
 	return NewWorkflowLease(wfContext, release, mutableState), nil
 }
 
+func (c *WorkflowConsistencyCheckerImpl) getCurrentWorkflowContext(
```

Comment:
Can this reuse the impl in getCurrentWorkflowContextWithConsistencyCheck? They are almost identical.

Reply:
in the first version idea was to keep the old code. 
I updated the PR, and try to reuse it.